projects/core/src/product/store/effects/product-search.effect.ts

```diff
@@ -35,14 +41,28 @@ export class ProductsSearchEffects {
                     action.auxiliary
                   );
                 }),
-                catchError((error) =>
-                  of(
+                catchError((error) => {
+                  const normalizedError: HttpErrorModel = tryNormalizeHttpError(
+                    error,
+                    this.logger
+                  );
+                  if (
+                    normalizedError?.details &&
+                    normalizedError.details[0].type ===
+                      'ArrayIndexOutOfBoundsError'
+                  ) {
```

Comment:
```suggestion
                  if (
                    normalizedError?.details?.[0].type ===
                      'ArrayIndexOutOfBoundsError'
                  ) {
```
Is there a reason not to continue using optional chaining operators?

Reply:
I've changed it, thanks for the suggestion